tests/e2e_runner.ts

```diff
@@ -0,0 +1,92 @@
+import { blue, bold, green, red, yellow, white } from 'chalk';
+import * as glob from 'glob';
+import * as path from 'path';
+import * as minimist from 'minimist';
+
+Error.stackTraceLimit = Infinity;
+
+const argv = minimist(process.argv.slice(2), {
+  boolean: ['debug', 'verbose'],
+  string: ['glob', 'ignore']
+});
+
+process.exitCode = 255;
+
+const testGlob = 'tests/**/*.ts';
+let currentFileName = null;
+let index = 0;
+
+const e2eRoot = path.join(__dirname, 'e2e');
+
+const allTests = glob.sync(path.join(e2eRoot, testGlob), { nodir: true, ignore: argv.ignore })
+  .map(name => path.relative(e2eRoot, name))
+  .sort();
```

Comment:
we will name test module as
```
000_user.ts
010_publish.ts
020_unpublish.ts
...
```
this `sort` ensures that this will run in right order.